packages/grafana-prometheus/src/components/monaco-query-field/monaco-completion-provider/completions.ts

```diff
@@ -176,19 +187,19 @@ async function getLabelValuesForMetricCompletions(
   }));
 }
 
-export async function getCompletions(situation: Situation, dataProvider: DataProvider): Promise<Completion[]> {
+export function getCompletions(situation: Situation, dataProvider: DataProvider): Promise<Completion[]> {
   switch (situation.type) {
     case 'IN_DURATION':
-      return DURATION_COMPLETIONS;
+      return Promise.resolve(DURATION_COMPLETIONS);
     case 'IN_FUNCTION':
       return getAllFunctionsAndMetricNamesCompletions(dataProvider);
     case 'AT_ROOT': {
       return getAllFunctionsAndMetricNamesCompletions(dataProvider);
     }
     case 'EMPTY': {
-      const metricNames = await getAllMetricNamesCompletions(dataProvider);
-      const historyCompletions = await getAllHistoryCompletions(dataProvider);
-      return [...historyCompletions, ...FUNCTION_COMPLETIONS, ...metricNames];
+      const metricNames = getAllMetricNamesCompletions(dataProvider);
+      const historyCompletions = getAllHistoryCompletions(dataProvider);
+      return Promise.resolve([...historyCompletions, ...FUNCTION_COMPLETIONS, ...metricNames]);
```

Comment:
Do you think we should use `Promise.all()` here?

Reply:
`getAllMetricNamesCompletions` and `getAllHistoryCompletions` used to be `async`, but aren't `async` anymore and do not return promises.